Automattic/wp-calypso
# Purchases: Use `state.purchases` to render `/purchases`

This is part of a series of PRs to replace `PurchasesStore` with a new branch of global state.

This PR updates the following components to use actions and data from `state/purchases/` instead of `PurchasesStore` and Flux actions:

- `PurchasesList`
- `ManagePurchase`
- `CancelPurchase`
- `CancelPrivateRegistration`
- `ConfirmCancelDomain`
- `RemovePurchase`
- `EditCardDetails`

It also removes the now unused actions and data components that were used in these components, and replaces an instance of `clearPurchases` in `Checkout`.

**Note:** This is large. ~80 lines are commits that will be handled separately, and will be removed before merging. I tried to make the revision history easy to read, and would recommend looking at the commits individually.

**Testing**
`PurchasesList`
- Visit `/purchases`.
- Assert that you see placeholders for a list of purchases.
- Assert these placeholders are eventually replaced with your actual purchases.

`ManagePurchase`
- Click on a purchase on `/purchases`.
- Assert that you are taken to a page with information about that purchase.
- Refresh the page.
- Assert that you see a loading placeholder.
- Assert that the placeholder is eventually replaced with information about the purchase.

`CancelPurchase`
- Visit `/purchases` and click on a renewable purchase that is outside the refund period.
- Click the `Cancel Subscription` link.
- Click the `Cancel Subscription` button.
- Assert that you are redirected to `/purchases` with a success notice and that the purchases are fetching again.

`CancelPrivateRegistration`
- Apply D2256-code and ensure that your hosts file points to your sandbox.
- Visit `/purchases` and click on a domain purchase with a private registration.
- Click the `Cancel Private Registration` link.
- Click the `Cancel Private Registration` button.
- Assert that you see an error message that reads `foo Please try again later or contact support.` (`foo` is just a generic string the API returns in my patch as an error message).
- Revert the patch.
- Click the `Cancel Private Registration` button again.
- Assert that you are redirected to `ManagePurchase` for this domain with a success notice.

`ConfirmCancelDomain`
- Visit `/purchases` and click on a domain purchase that is within the refund period.
- Click the `Cancel Domain and Refund` link.
- Click the `Cancel Domain and Refund` button.
- Select a reason to cancel and click `Cancel Anyway`.
- Assert that you are redirected to `/purchases` with a success notice and that the purchases are fetching again.

`RemovePurchase`
- Visit `/purchases` and click on a purchase that has auto-renew disabled.
- Click `Remove [purchase name]`.
- Fill out the survey and proceed through the dialog.
- Assert that you are redirected to the list of purchases, and that this purchase is not in that list.

`EditCardDetails`
- Visit `/purchases` and click on a purchase made with a credit card.
- Click `Edit Payment Method`.
- Assert that you see a placeholder as the list of the user's stored cards fetch.
- Assert that the page eventually loads.
- Enter and submit new credit card details.
- Assert that you are redirected to `ManagePurchase` for the given purchase, and that you see placeholders while the purchase is fetched again.

`Checkout`
- Visit `/purchases` and wait until the purchases have loaded.
- Purchase a plan in the same browser session (by clicking through `My Sites`, `Plan`, etc.).
- Once you reach the "thank you" page in checkout, navigate to `/purchases` in the same session.
- Assert that you see placeholders as the purchases load again, and that the purchase you just made appears in the list of purchases.
